src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -1047,11 +1060,14 @@ namespace cryptonote
 
     // the tx hash is the hash of the 3 hashes
     crypto::hash res = cn_fast_hash(hashes, sizeof(hashes));
+    t.set_hash(res);
```

Comment:
Indeed. However, since this is a drive by "let's set this since it's a cheap extra step", I'd rather keep it in core::handle_incoming_tx_pre since removing this set_hash would still be a correct implementation of get_pruned_transaction_hash.